persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/JdbcRepositoryHelper.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.justice.services.jdbc.persistence;
+
+import static uk.gov.justice.services.jdbc.persistence.PreparedStatementWrapper.valueOf;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.sql.DataSource;
+
+public class JdbcRepositoryHelper<T> {
+
+    public PreparedStatementWrapper preparedStatementWrapperOf(final DataSource dataSource, final String query) throws SQLException {
+        return valueOf(dataSource.getConnection(), query);
+    }
+
+    public Stream<T> streamOf(final PreparedStatementWrapper psWrapper, final Function<ResultSet, T> function) throws SQLException {
+
+        final ResultSet resultSet = psWrapper.executeQuery();
+
+        return internalStreamOf(psWrapper, resultSet, function);
+    }
+
+    /**
+     * @param psWrapper               prepared statement wrapper
+     * @param resultSet               jdbc resultSet
+     * @param resultSetToEntityMapper - function mapping resultSet to entity of type U
+     * @param <U>                     - generic type of entity
+     * @return stream of entities
+     */
+    public <U> Stream<U> spliteratorStreamOf(final PreparedStatementWrapper psWrapper, final ResultSet resultSet, final Function<ResultSet, U> resultSetToEntityMapper) {
+        return StreamSupport.stream(new Spliterators.AbstractSpliterator<U>(
```

Comment:
Can this be an abstract class? Is it slightly cleaner?

Reply:
Separate ticket around refactoring that